benchmark/framework/cli.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Cli implementation for performance tests suits."""
+import ast
+import inspect
+from typing import Any, Callable, Dict, List, Optional
+
+import click
+from click.core import Argument, Command, Context, Option, Parameter
+
+from .executor import Executor
+from .func_details import FuncDetails
+from .report_printer import PerformanceReportPrinter
+
+
+class DefaultArgumentsMultiple(Argument):
+    """Multiple arguments with default value."""
+
+    def __init__(self, *args, **kwargs):
+        """Create MultipleArguments instance."""
+        kwargs["nargs"] = -1
+        default = kwargs.pop("default", tuple())
+        super().__init__(*args, **kwargs)
+        self.default = default
+
+    def full_process_value(self, ctx: Context, value: Any) -> Any:
+        """Given a value and context this runs the logic to convert the value as necessary."""
+        return super().process_value(ctx, value or self.default)
+
+
+class TestCli:
+    """Performance test client."""
+
+    def __init__(
+        self,
+        func: Callable,
+        executor_class=Executor,
+        report_printer_class=PerformanceReportPrinter,
+    ):
+        """
+        Make performance client.
+
+        func - should be function with first parameter multithreading.Queue.
+        func  - should have docstring, and default values for every extra argument.
```

Comment:
It would be good if there was a short example here in the docstring how this is intended to be used.

Reply:
done